### PR TITLE
chore(csharp): Abundant Number

### PR DESCRIPTION
**Pull request by UG-SEP**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] New algorithm
- [ ] Optimization in previous algorithms
- [ ] Code style update
- [ ] Refactor
- [ ] Documentation
- [ ] Other, please describe:

**Briefly describe the changes in this PR**
Created a folder name: `Maths` in CSharp/src dir and added the solution.cs there
**Other information:**

closes: #355

### CODE REVIEW

File: algorithms/CSharp/src/Maths/abundant-number.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
+/* Program to check whether a number is abundant or not */
+using System;
+
+namespace abundant_number
+{
+    class Solution
+    { 
+        // function to check whether number is abundant or not
+        public static bool Abudant(int num)
+    {
```
Comment (ming-tsai) [inline algorithms/CSharp/src/Maths/abundant-number.cs:10]: Please fix the ident
```suggestion
        {
```

File: algorithms/CSharp/src/Maths/abundant-number.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
+/* Program to check whether a number is abundant or not */
+using System;
+
+namespace abundant_number
+{
+    class Solution
+    { 
+        // function to check whether number is abundant or not
+        public static bool Abudant(int num)
```
Comment (ming-tsai) [inline algorithms/CSharp/src/Maths/abundant-number.cs:9]: Rename the method name to `IsAbudant`
```suggestion
        public static bool IsAbudant(int num)
```

File: algorithms/CSharp/src/Maths/abundant-number.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+                // if the remainder becomes 0 add that number to sum
+                if (num % i == 0)
+                    sum += i;
+            }
+            // if the sum is greater than num that means it is an abundant number
+            if (sum > num)
+                return true;
+            // it is not an abundant number
+            else
+                return false;
```
Comment (ming-tsai) [inline algorithms/CSharp/src/Maths/abundant-number.cs:24]: Add the missing curly brackets
```suggestion
            if (sum > num)
            {
                return true;
            }
            else // it is not an abundant number
            {
                return false;
            }
```

File: algorithms/CSharp/src/Maths/abundant-number.cs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+            int num;
+            bool res;
+            Console.WriteLine("Enter a number: ");
+            // reading data and converting into into int to store that in num variable
+            num = Convert.ToInt32(Console.ReadLine());
+            res = Abudant(num);
+            if (res)
+                Console.WriteLine($"{num} is an abundant number");
+            else
+                Console.WriteLine($"{num} is not an abundant number");
```
Comment (ming-tsai) [inline algorithms/CSharp/src/Maths/abundant-number.cs:38]: Add the missing curly brackets

**Review by ming-tsai — CHANGES_REQUESTED** [review]
Hi @UG-SEP,
I just add some comments about the coding style, please add a unit test for this and ensure this works on [dotnetfiddle NET 5](https://dotnetfiddle.net/).

**Comment by UG-SEP** [comment]
> Hi @UG-SEP,
> I just add some comments about the coding style, please add a unit test for this and ensure this works on [dotnetfiddle NET 5](https://dotnetfiddle.net/).

@ming-tsai I need to add a test unit in the test folder 
Am I right

**Comment by ming-tsai** [comment]
> > Hi @UG-SEP,
> > I just add some comments about the coding style, please add a unit test for this and ensure this works on [dotnetfiddle NET 5](https://dotnetfiddle.net/).
> 
> @ming-tsai I need to add a test unit in the test folder 
> Am I right
> 

Yes

**Comment by UG-SEP** [comment]
Problem
@ming It is not showing to add a reference to the Abundant. cs well I have created this in the same project and also Installed NUnit & NUnit3 test adapter 
Can you please help me

**Comment by ming-tsai** [comment]
> Problem
> @Ming It is not showing to add a reference to the Abundant. cs well I have created this in the same project and also Installed NUnit & NUnit3 test adapter
> Can you please help me

You could guide with [Palindrome test on C#](https://github.com/MakeContributions/DSA/blob/main/algorithms/CSharp/test/Strings/palindrome.cs)

**Comment by UG-SEP** [comment]
unittest
All test cases pass successfully
Thanks, @ming-tsai  for helping me during the unit test work
I am too glad to have you with me...

**Review by ming-tsai — APPROVED** [review]

**Comment by ming-tsai** [comment]
Nice work

**Pull request merged**